integration-tests/jest/jest-utils.js

```diff
@@ -27,4 +35,63 @@ const registerTest = ({ fn, name, timeout = defaultTimeout, targetEnvironments =
     timeout)
 }
 
-module.exports = { registerTest }
+const flakeShaker = ({ fn, name }) => {
+  const timeoutMillis = clusterTimeout * 60 * 1000
+  const padding = 100
+  const messages = ['', `Number of times to run this test: ${testRuns} (adjust this by setting RUNS in your environment)`,
+    '',
+    `Number of concurrent test runs: ${maxConcurrency} (adjust this by setting CONCURRENCY in your environment)`,
+    '',
+    `Timeout (minutes): ${clusterTimeout} (if your test times out adjust CLUSTER_TIMEOUT_MINUTES in your environment)`,
+    '']
+
+  rawConsole.log(`\n\x1b[1m${/* bold */ '╔'.padEnd(padding, '═')}╗`)
+  rawConsole.log(`${`║ Running flake shaker on ${name} to flush out test flakiness...`.padEnd(padding)}║`)
+  rawConsole.log(`${'╚'.padEnd(padding, '═')}╝`)
+
+  const message = _.flow(
+    _.map(line => `${`║ ${line}`.padEnd(padding)}║\n`),
+    list => [...list, `${'╚'.padEnd(padding, '═')}╝`],
+    _.join(''))(messages)
+  rawConsole.log(`${message}`)
+
+  const runCluster = async () => {
+    const cluster = await Cluster.launch({
+      concurrency: Cluster.CONCURRENCY_CONTEXT,
+      maxConcurrency: _.toInteger(maxConcurrency),
+      timeout: timeoutMillis,
+      puppeteerOptions: {
+        defaultViewport: { width: 1200, height: 800 }
+      }
+    })
+
+    await cluster.task(async ({ page, data }) => {
+      const { taskFn, taskParams, runId } = data
+      try {
+        const result = await taskFn({ context, page, ...taskParams })
+        rawConsole.log(`Test number ${runId} passed`)
+        return result
+      } catch (e) {
+        rawConsole.log(`Test number ${runId} failed: ${e}`)
+        return e
+      }
+    })
+
+    const runs = _.times(runId => cluster.execute({ taskParams: targetEnvParams, taskFn: fn, runId }), testRuns)
+    const results = await Promise.all(runs)
+    const errors = _.filter(_.isError, results)
+    const numErrors = _.size(errors)
+
+    await cluster.idle()
+    await cluster.close()
+
+    if (numErrors) {
```

Comment:
I think we suggest adding `!!` when using a non-boolean as predicate for explicitness.
```suggestion
    if (!!numErrors) {
```